sdks/python/apache_beam/io/bigquery.py

```diff
@@ -37,13 +37,14 @@
 https://cloud.google.com/bigquery/bq-command-line-tool-quickstart
 
 BigQuery sources can be used as main inputs or side inputs. A main input
-(common case) is expected to be massive and the Dataflow service will make sure
-it is split into manageable chunks and processed in parallel. Side inputs are
-expected to be small and will be read completely every time a ParDo DoFn gets
-executed. In the example below the lambda function implementing the DoFn for the
-Map transform will get on each call *one* row of the main table and *all* rows
-of the side table. The execution framework may use some caching techniques to
-share the side inputs between calls in order to avoid excessive reading::
+(common case) is expected to be massive and the execution framework will make
```

Comment:
you are talking about what execution engine will do?
if you are using native source, it is runner-specific.
if not, then it is the source guaranteeing that.